Purchases/Purchasing/IntroEligibilityCalculator.swift

```diff
@@ -94,10 +95,13 @@ private extension IntroEligibilityCalculator {
                     return foundByGroupId
                 }
 
-            let hasIntroductoryPrice = candidate.introductoryDiscount != nil
-            result[candidate.productIdentifier] = !hasIntroductoryPrice || usedIntroForProductIdentifier
-                ? IntroEligibilityStatus.ineligible
-                : IntroEligibilityStatus.eligible
+            if candidate.introductoryDiscount == nil {
+                result[candidate.productIdentifier] = .noIntroOfferExists
```

Comment:
>If you've set up introductory prices in App Store Connect, the introductory price property will be populated. This property is nil if the product has no introductory price.

Taken from https://developer.apple.com/documentation/storekit/skproduct/2936878-introductoryprice

Reply:
Thank you for pointing this documentation out 😄